pennylane/optimize/qng.py

```diff
@@ -116,17 +116,27 @@ def step(self, qnode, x, recompute_tensor=True):
             recompute_tensor (bool): Whether or not the metric tensor should
                 be recomputed. If not, the metric tensor from the previous
                 optimization step is used.
+            metric_tensor_fn (function): Optional metric tensor function
+                with respect to the variables ``x``.
+                If ``None``, the metric tensor function is computed automatically.
 
         Returns:
             array: the new variable values :math:`x^{(t+1)}`
         """
         # pylint: disable=arguments-differ
-        if not hasattr(qnode, "metric_tensor"):
-            raise ValueError("Objective function must be encoded as a single QNode")
+        if not hasattr(qnode, "metric_tensor") and not metric_tensor_fn:
+            raise ValueError(
+                "The objective function must either be encoded as a single QNode or "
+                "a VQECost object for the natural gradient to be automatically computed.
```

Comment:
Oops, had a typo in my suggestion.
```suggestion
                "a VQECost object for the natural gradient to be automatically computed. "
```